src/Services/DatabaseAuditLog.cs

```diff
@@ -0,0 +1,46 @@
+using System.Security.Claims;
+using IdentityServerConfig.Data;
+using IdentityServerConfig.Models;
+using Microsoft.AspNetCore.Components.Authorization;
+
+namespace IdentityServerConfig.Services;
+
+public class DatabaseAuditLog : IAuditLog
+{
+    private readonly AuditContext _auditContext;
+    private readonly AuthenticationStateProvider _authenticationStateProvider;
+    private readonly ILogger _logger;
+
+    public DatabaseAuditLog(AuditContext auditContext, AuthenticationStateProvider authenticationStateProvider, ILogger<DatabaseAuditLog> logger)
+    {
+        _auditContext = auditContext;
+        _authenticationStateProvider = authenticationStateProvider;
+        _logger = logger;
+    }
+    
+    public void Log(string action, IDictionary<string,string> data)
+    {
+        var user = _authenticationStateProvider.GetAuthenticationStateAsync().Result.User;
```

Comment:
Using `.Result` on a Task could result in a deadlock. We should probably make this method asynchronous and use `await` here to avoid that.

Reply:
fixed it in 5a020fb